### PR TITLE
Fill control origin

### PR DESCRIPTION
**Pull request by jcscottiii**
Builds on https://github.com/opencontrol/fedramp-templater/pull/42. Delivers https://github.com/opencontrol/fedramp-templater/issues/26.

This PR introduces filling of the control origination checkbox based on the values in the opencontrol yaml. 

The logic for unchecking is there but the logic for WHEN to uncheck will be addressed in #45

### CODE REVIEW

File: control/control_origination.go [hunk 1]
```diff
@@ -59,7 +59,27 @@ func detectControlOriginKey(textNodes []xml.Node) string {
...
+	case "customer_configured":
+		return configuredByCustomerOrigination
+	case "customer_provided":
+		return providedByCustomerOrigination
+	case "shared":
+		return sharedOrigination
+	case "inherited":
+		return inheritedOrigination
+	}
+	return noOrigin
```
Comment (afeld) [inline control/control_origination.go]: Can we simplify this and the variables above to simply be a map from the origin key to the document substring?

File: control/summary_table.go [hunk 2]
```diff
@@ -16,29 +16,62 @@ type SummaryTable struct {
 }
 
 // NewSummaryTable creates a SummaryTable instance.
-func NewSummaryTable(root xml.Node) SummaryTable {
+func NewSummaryTable(root xml.Node) *SummaryTable {
 	tbl := table{Root: root}
-	return SummaryTable{tbl}
+	return &SummaryTable{tbl}
```
Comment (afeld) [inline control/summary_table.go]: My inclination is to stick to passing by value whenever possible...mind trying to change this back, and seeing if it still works?

File: control/summary_table.go [hunk 3]
```diff
@@ -16,29 +16,62 @@ type SummaryTable struct {
...
 	}
 
-	control, err := st.controlName()
+	roles := openControlData.GetResponsibleRoles(control)
+	roleCell.setValue(roles)
+	return
+}
+
+
+
```
Comment (afeld) [inline control/summary_table.go]: Very minor nitpick: mind making the spacing between functions consistent?

File: control/summary_table.go [hunk 4]
```diff
@@ -16,29 +16,62 @@ type SummaryTable struct {
...
+	controlOrigins := openControlData.GetControlOrigins(control)
+	for _, controlOrigin := range controlOrigins {
+		controlOriginKey := detectControlOriginKeyFromData(controlOrigin)
+		if controlOriginKey == noOrigin {
+			continue
+		}
+		controlOrigination.origins[controlOriginKey].SetCheckMarkTo(true)
+	}
+
+	// TODO. ensure the ones, not specified are set to false.
```
Comment (afeld) [inline control/summary_table.go]: Mind moving to an issue?

**Comment by afeld** [comment]
All looks good! Just a couple small things.

File: control/summary_table.go (hunk 2 above)
Comment (jcscottiii) [reply control/summary_table.go]: done

File: control/summary_table.go (hunk 3 above)
Comment (jcscottiii) [reply control/summary_table.go]: done

File: control/summary_table.go (hunk 4 above)
Comment (jcscottiii) [reply control/summary_table.go]: https://github.com/opencontrol/fedramp-templater/issues/45

File: control/control_origination.go [hunk 5]
```diff
@@ -1,41 +1,76 @@
...
-		serviceProviderSystemSpecificOrigination,
-		serviceProviderHybridOrigination,
-		configuredByCustomerOrigination,
-		providedByCustomerOrigination,
-		sharedOrigination,
-		inheritedOrigination,
+type originMapping struct {
+	yamlMapping string
+	docMapping  string
+}
```
Comment (afeld) [inline control/control_origination.go:28]: Nice solution!

**Pull request merged**